native-maven-plugin/src/main/java/org/graalvm/buildtools/maven/NativeCompileNoForkMojo.java

```
@@ -113,78 +111,29 @@ public void execute() throws MojoExecutionException {
-                        "An SBOM will be embedded in the image.", sbomNativeImageFlag, AUGMENTED_SBOM_PARAM_NAME));
-            }
-
-            /* Continue to generate augmented SBOM because parameter option explicitly set and all conditions are met. */
-        } else {
-            if (!isOracleGraalVM(logger) || !sbomEnabledForNativeImage) {
-                return;
-            }
-
-            if (!SBOMGenerator.checkAugmentedSBOMSupportedByJDKVersion(detectedJDKVersion, false)) {
-               return;
-            }
-
-            /*
-             * Continue to generate augmented SBOM because although the parameter option was not set, SBOM is used for
-             * Native Image and all conditions are met.
-             */
+    private void generateBaseSBOMIfNeeded() throws MojoExecutionException {
+        if (skipBaseSBOM) {
+            logger.info(String.format("Skipping base SBOM generation (parameter '%s' is true).", SKIP_BASE_SBOM_PARAM_NAME));
+            return;
         }
 
-        var sbomGenerator = new SBOMGenerator(mavenProject, mavenSession, pluginManager, repositorySystem, mainClass, logger);
+        var sbomGenerator = new SBOMGenerator(mavenProject, mavenSession, pluginManager, repositorySystem, mainClass);
+        var config = new SBOMGenerator.Config(isOracleGraalVM(logger), getBuildArgs(), NativeImageUtils.getMajorJDKVersion(getVersionInformation(logger)));
         try {
-            sbomGenerator.generate();
-        } catch (MojoExecutionException e) {
-            /* Only throw exception for users that explicitly opt-in to using augmented SBOMs. */
-            if (optionWasSet) {
-                throw e;
-            }
-            logger.warn(String.format("Could not generate an augmented SBOM: %s. Fallback to generating a non-augmented SBOM.",
+            sbomGenerator.generateIfSupportedAndEnabled(config);
+        } catch (Exception e) {
```

We now catch all exceptions to avoid the risk of a runtime exception bypassing this fallback mechanism.

---

I agree, that's better. I'll update accordingly.